File: solutions/c/armstrong-numbers/1/armstrong_numbers.c
#include "armstrong_numbers.h"
#include <math.h>

bool is_armstrong_number(int candidate) {
    int remainder = candidate;
    int sum = 0;
    int power = (int)ceil(log10(candidate));
    while (remainder > 0) {
        int digit = remainder % 10;
        sum += (int)pow(digit, power);
        remainder /= 10;
    }
    return sum == candidate;
}

File: solutions/go/armstrong-numbers/1/armstrong_numbers.go
package armstrong

import "math"

func IsNumber(n int) bool {
    numDigits := math.Ceil(math.Log10(float64(n)))
    sum := float64(n)
    for n > 0 {
        sum -=  math.Pow(float64(n % 10), numDigits)
        n /= 10
    }
    return sum == 0
}
